tests/TestCase/Database/Type/DateTimeFractionalTypeTest.php

```diff
@@ -0,0 +1,396 @@
+
+        $tz = $date->getTimezone();
+        $this->type->setTimezone('Asia/Kolkata'); // UTC+5:30
+        $result = $this->type->toDatabase($date, $this->driver);
+        $this->assertSame('2013-08-12 20:46:17.123456', $result);
+        $this->assertEquals($tz, $date->getTimezone());
+
+        $this->type->setTimezone(new DateTimeZone('Asia/Kolkata'));
+        $result = $this->type->toDatabase($date, $this->driver);
+        $this->assertSame('2013-08-12 20:46:17.123456', $result);
+        $this->type->setTimezone(null);
+
+        $date = new FrozenTime('2013-08-12 15:16:17.123456');
+        $result = $this->type->toDatabase($date, $this->driver);
+        $this->assertSame('2013-08-12 15:16:17.123456', $result);
+
+        $this->type->setTimezone('Asia/Kolkata'); // UTC+5:30
+        $result = $this->type->toDatabase($date, $this->driver);
+        $this->assertSame('2013-08-12 20:46:17.123456', $result);
+        $this->type->setTimezone(null);
+    }
+
+    /**
+     * Test converting to database format without microseconds
+     *
+     * @return void
+     */
+    public function testToDatabaseNoMicroseconds()
+    {
+        $date = new Time('2013-08-12 15:16:17');
+        $result = $this->type->toDatabase($date, $this->driver);
+        $this->assertSame('2013-08-12 15:16:17.000000', $result);
+
+        $tz = $date->getTimezone();
+        $this->type->setTimezone('Asia/Kolkata'); // UTC+5:30
+        $result = $this->type->toDatabase($date, $this->driver);
+        $this->assertSame('2013-08-12 20:46:17.000000', $result);
+        $this->assertEquals($tz, $date->getTimezone());
+
+        $this->type->setTimezone(new DateTimeZone('Asia/Kolkata'));
+        $result = $this->type->toDatabase($date, $this->driver);
+        $this->assertSame('2013-08-12 20:46:17.000000', $result);
+        $this->type->setTimezone(null);
+
+        $date = new FrozenTime('2013-08-12 15:16:17');
+        $result = $this->type->toDatabase($date, $this->driver);
+        $this->assertSame('2013-08-12 15:16:17.000000', $result);
+
+        $this->type->setTimezone('Asia/Kolkata'); // UTC+5:30
+        $result = $this->type->toDatabase($date, $this->driver);
+        $this->assertSame('2013-08-12 20:46:17.000000', $result);
+        $this->type->setTimezone(null);
+
+        $date = 1401906995;
+        $result = $this->type->toDatabase($date, $this->driver);
+        $this->assertSame('2014-06-04 18:36:35.000000', $result);
+    }
+
+    /**
+     * Data provider for marshal() with microseconds
+     *
+     * @return array
+     */
+    public function marshalProvider()
+    {
+        return [
+            // invalid types.
+            [null, null],
+            [false, null],
+            [true, null],
+            ['', null],
+            ['derpy', null],
+            ['2013-nope!', null],
+
+            // valid string types
+            ['2014-02-14 00:00:00.123456', new FrozenTime('2014-02-14 00:00:00.123456')],
```

Comment:
Could we have an invalid fractional seconds? Like 7 or 8 points?

Reply:
Yes and no.  In the marshal() format, it will throw an error on extra trailing data.  On the toPHP() format, it will silently ignore them.

I'm hoping to get rid of the strict formatting in the future, but I can add the failure case for marshal and the success case to toPHP.

Reply:
I should note that the original DateTimeType would have failed in the same way essentially.